src/init.cc

```diff
@@ -1241,11 +1224,14 @@ static ncclResult_t initTransportsRank(struct ncclComm* comm, struct ncclComm* p
   const char* dumpXmlFile;
   dumpXmlFile = ncclGetEnv("NCCL_TOPO_DUMP_FILE");
   if (dumpXmlFile) {
-    NCCLCHECKGOTO(ncclTopoGetSystem(comm, NULL, dumpXmlFile), ret, fail);
+    //NCCLCHECKGOTO(ncclTopoGetSystem(comm, NULL, dumpXmlFile), ret, fail);
+    NCCLCHECKGOTO(ncclTopoGetSystemNew(comm, &comm->topo, dumpXmlFile), ret, fail);
+  }else{
+    NCCLCHECKGOTO(ncclTopoGetSystemNew(comm, &comm->topo, NULL), ret, fail);
   }
 
   // Topo detection / System graph creation
-  NCCLCHECKGOTO(ncclTopoGetSystem(comm, &comm->topo), ret, fail);
+  //NCCLCHECKGOTO(ncclTopoGetSystem(comm, &comm->topo), ret, fail);
```

Comment:
```suggestion

  // Topo detection / System graph creation
  NCCLCHECKGOTO(ncclTopoGetSystemNew(comm, &comm->topo, dumpXmlFile), ret, fail);
```
This should be equivalent. I'm wondering, is `ncclTopoGetSystem` called anywhere now? If so, why not modify it instead of adding `ncclTopoGetSystemNew`?